transmogrifier/sources/datacite.py

```diff
@@ -188,59 +201,62 @@ def create_from_datacite_xml(
                 kind=identifier_xml["identifierType"],
             ),
         ]
-        alternate_identifiers = xml.metadata.find_all("alternateIdentifier")
-        for alternate_identifier in alternate_identifiers:
-            i = timdex.Identifier(
-                value=alternate_identifier.string,
+        for alternate_identifier in [
+            i for i in xml.metadata.find_all("alternateIdentifier") if i.string
+        ]:
+            kwargs["identifiers"].append(
+                timdex.Identifier(
+                    value=alternate_identifier.string,
+                    kind=alternate_identifier.get("alternateIdentifierType"),
+                )
             )
-            if "alternateIdentifierType" in alternate_identifier.attrs:
-                i.kind = alternate_identifier.attrs["alternateIdentifierType"]
-            kwargs["identifiers"].append(i)
 
         related_identifiers = xml.metadata.find_all("relatedIdentifier")
         for related_identifier in [
             i
             for i in related_identifiers
-            if "relationType" in i.attrs and i.attrs["relationType"] == "IsIdenticalTo"
+            if i.get("relationType") == "IsIdenticalTo" and i.string
         ]:
             i = timdex.Identifier(
                 value=cls.generate_related_item_identifier_url(related_identifier),
-                kind=related_identifier.attrs["relationType"],
+                kind=related_identifier.get("relationType"),
             )
             kwargs["identifiers"].append(i)
 
         # language
         language = xml.metadata.find("language")
-        if language:
+        if language and language.string:
             kwargs["languages"] = [language.string]
 
         # locations
```

Comment:
We might consider adding the Digital object URL link here since we have it in the other two transforms. In this case I think it would be the same as the source link unless you can see a different field to pull from. (Could be a new ticket)

Reply:
I added this but there was discussion in the retro about whether it was useful to have duplicate values here so we might discuss this with the team